docs/user-defined-type-functions.md

```diff
@@ -0,0 +1,235 @@
+| `getnumber()` | `typelib` | returns an immutable runtime representation of the built-in type `number` |
+| `getstring()` | `typelib` | returns an immutable runtime representation of the built-in type `string` |
+| `getstringsingleton(arg: string)` | `typelib` | returns an immutable runtime representation of a string singleton type of the argument |
+| `getbooleansingleton(arg: boolean)` | `typelib` | returns an immutable runtime representation of a boolean singleton type of the argument |
+| `getunion(arg: {typelib})` | `typelib` | returns an immutable runtime representation of union type of its argument |
+| `getintersection(arg: {typelib})` | `typelib` | returns an immutable runtime representation of intersection type of its argument |
+| `newtable()` | `typelib` | returns a mutable runtime representation of a `table` type |
+| `newmetatable()` | `typelib` | returns a mutable runtime representation of `Metatable` |
+| `newfunction()` | `typelib` | returns a mutable runtime representation of a `function` type |
+| `isnil()` | `boolean` | returns true if self is a runtime representation of the built-in type `nil` |
+| `isunknown()` | `boolean` | returns true if self is a runtime representation of the built-in type `unknown` |
+| `isnever()` | `boolean` | returns true if self is a runtime representation of the built-in type `never` |
+| `isany()` | `boolean` | returns true if self is a runtime representation of the built-in type `any` |
+| `isnegation()` | `boolean` | returns true if self is a runtime representation of a `Negation` |
+| `isboolean()` | `boolean` | returns true if self is a runtime representation of the built-in type`boolean` |
+| `isnumber()` | `boolean` | returns true if self is a runtime representation of the built-in type `number` |
+| `isstring()` | `boolean` | returns true if self is a runtime representation of the built-in type `string` |
+| `isstringsingleton()` | `boolean` | returns true if self is a runtime representation of a string singleton |
+| `isbooleansingleton()` | `boolean` | returns true if self is a runtime representation of a boolean singleton |
+| `isunion()` | `boolean` | returns true if self is a runtime representation of the union type |
+| `isintersection()` | `boolean` | returns true if self is a runtime representation of the intersection type |
+| `istable()` | `boolean` | returns true if self is a runtime representation of a `table` type |
+| `ismetatable()` | `boolean` | returns true if self is a runtime representation of `Metatable` |
+| `isfunction()` | `boolean` | returns true if self is a runtime representation of a `function` type |
+| `isclass()` | `boolean` | returns true if self is a runtime representation of a `class` type |
+| `issubtypeof(arg: typelib)` | `boolean` | returns true if self is a subtype or equal to arg in the type hierarchy |
+| `conformsto(arg: typelib)` | `boolean` | returns true if self is equal to arg in the type hierarchy |
```

Comment:
I don't understand what this function means at all. `conformsto`?

Reply:
Yeah. Thought we agreed to using `__eq` and `__le` for subtyping.

Reply:
> I don't understand what this function means at all. conformsto?

ah, `conformsto` will be change to `equalsto`.

> Thought we agreed to using __eq and __le for subtyping.

I remember in one of the meetings, Andy preferred to have them as methods rather than overloading operators. We could do both?

Reply:
I think I would like to pick one or the other. I suggested `__eq` and `__le` because it felt intuitive. We already say things like `T <: U` which can be expressed in Luau terms as `t <= u`.